pyproject.toml

```diff
@@ -0,0 +1,152 @@
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: System :: Archiving :: Packaging",
+    "Topic :: Utilities",
+]
+dependencies = [ 
+    "conda-lock>=1.2",
+    "lockfile",
+    "ruamel.yaml",
+    "pydantic",
+]
+
+[project.optional-dependencies]
+docs_requirements = [
+    "Sphinx>=5.1.1",
+    "sphinx-autobuild>=2021.3.14",
+    "sphinx-autodoc-typehints>=1.19.2",
+    "myst-parser>=0.18.0",
+    "pydata-sphinx-theme>=0.9.0",
+]
+
+[project.scripts]
+console_scripts = "conda-project=conda_project.cli.main:main"
+
+[project.urls]
+repository = "https://github.com/conda-incubator/conda-project"
+homepage = "https://github.com/conda-incubator/conda-project"
+documentation = "https://conda-incubator.github.io/conda-project/user_guide.html"
+"Issue Tracker" = "https://github.com/conda-incubator/conda-project/issues"
+#"Conda-Forge Feedstock" = "https://github.com/conda-forge/conda-project-feedstock"
+#"AnacondaRecipes Feedstock" = "https://github.com/AnacondaRecipes/conda-project-feedstock"
+
+[tool]
+
+[tool.check-manifest]
+ignore = [
+    "*.yml",
+    ".coveragerc",
+    "Makefile",
+    "docs",
+    "docs/*",
+    "*.enc",
+    "tests",
+    "tests/*",
+]
+
+[tool.coverage.run]
+source = [
+    "conda_project/*",
+]
+
+[tool.coverage.report]
+exclude_lines = [
+    'if __name__ == "__main__":'
+]
+
+[tool.black]
+line-length = 79
+target-version = ['py37', 'py38', 'py39', 'py310', 'py311']
+#skip-string-normalization = false
+#skip-magic-trailing-comma = false
+
+[tool.isort]
+profile = "black"
+src_paths = ["."]
+#sections = FUTURE, STDLIB, THIRDPARTY, FIRSTPARTY, LOCALFOLDER
+default_section = "LOCALFOLDER"
+#known_third_party = numpy, pandas, keras, tensorflow, sklearn, matplotlib, scipy
+multi_line_output = 3
+lines_after_imports = 2
+force_single_line = true
+use_parentheses = true
+ensure_newline_before_comments = true
+line_length = 80
+
+[tool.flake8]
+max-complexity = 30
+max-line-length = 120
+#exclude = .git,__pycache__,docs,old,build,dist,build,conda_project/_version.py,tests,conda.recipe,.git,versioneer.py,benchmarks,.asv
+#ignore=E122,E123,E126,E127,E128,E731,E722,W503
+
+[tool.mypy]
+# 3rd party import
+ignore_missing_imports = true
+# dynamic typing
+disallow_any_unimported = true
+disallow_any_expr = false
+disallow_any_decorated = false
+disallow_any_explicit = true
+disallow_any_generics = false
+disallow_subclassing_any = true
+# platform
+python_version = "3.8"
+# untyped
+disallow_untyped_calls = true
+disallow_untyped_defs = true
+disallow_incomplete_defs = true
+disallow_untyped_decorators = true
+# None and Optional
+no_implicit_optional = true
+# Warnings
+warn_return_any = false
+warn_no_return = true
+warn_unreachable = true
+warn_unused_configs = true
+# Misc
+pretty = true
+follow_imports = "silent"
+check_untyped_defs = false
+strict_optional = false
+exclude = ["scripts", "docs", "test"]
```

Comment:
This is what caused the pre-commit bot to refactor the code. Can we please comment these out for now and handle tooling config changes in a separate PR?

Reply:
Not that I'm opposed to having clearer settings, I'd just like to do it more gradually in atomic PRs